### PR TITLE
Toz 163 securing api with roles

### PR DESCRIPTION
**Pull request by Jakub-Izbicki**

### CODE REVIEW

File: src/main/java/com/intive/patronage/toz/schedule/ScheduleController.java [hunk 1]
```diff
@@ -39,12 +40,20 @@
...
 
-    @ApiOperation("Get schedule")
+    @Autowired
+    ScheduleController(ScheduleService scheduleService, ScheduleParser scheduleParser) {
+        this.scheduleService = scheduleService;
+        this.scheduleParser = scheduleParser;
+    }
+
+    @ApiOperation(value = "Get schedule", notes =
+            "Required roles: SA, TOZ.")
```
Comment (MateuszSz93) [inline src/main/java/com/intive/patronage/toz/schedule/ScheduleController.java]: Shouldn't be added here (and in all other "@ApiOperation" in this controller) 'VOLUNTEER'?

File: src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        properties = {ApiProperties.JWT_SECRET_BASE64, ApiProperties.SUPER_ADMIN_PASSWORD}
+)
+@ActiveProfiles("test")
+public class TokenRoleTest {
+
+    private final static String TYPE_PARAM = "type";
+    private final static String RELEASED = "RELEASED";
+    private final static String ACHIEVED = "ACHIEVED";
+    private static final String EMAIL_VOLUNTEER = "volunteer@mail.com";
+    private final static User.Role ROLE_VOLUNTEER = User.Role.VOLUNTEER;
```
Comment (maras93) [inline src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java]: It's a cosmentic change but would be nice if you be consistent with others definitions of static and final variables. So please change it to `private static final ....`.

File: src/main/java/com/intive/patronage/toz/schedule/ScheduleController.java (hunk 1 above)
Comment (Jakub-Izbicki) [inline src/main/java/com/intive/patronage/toz/schedule/ScheduleController.java]: You're right, I didn't notice this. Thanks for mentioning that.

File: src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+    private User admin;
+    private User volunteer;
+
+    @Before
+    public void setUp() throws Exception {
+        final String passwordHash = passwordEncoder.encode(PASSWORD);
+        admin = new User();
+        volunteer = new User();
```
Comment (maras93) [inline src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java]: Why you didn't use data provider for each test? I think there is no special needs for creating `admin` and `volunteer`  before each test.

File: src/main/resources/i18n/messages.properties [hunk 4]
```diff
@@ -6,3 +6,4 @@ mustHaveValue=Attribute "{0}" must have a value among: {1}.
 invalidArgumentValue=Invalid argument value!
 invalidScheduleHours=Hours {0} are invalid! Valid hours for {1} are: {2}
 reservationAlreadyExists=Reservation at hour: {0}, day: {1}, already exists!
+accessDenied=Insufficient authority to access this resource! Your role/s are: {0}.
```
Comment (LukaszPiskadlo) [inline src/main/resources/i18n/messages.properties]: I think it's enough to say, that user has insufficient authority, we don't have to tell him his roles. But it's not necessarily an error.

File: src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java (hunk 3 above)
Comment (Jakub-Izbicki) [inline src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java]: Right. I dont even have to add the to db :) I will change that.

**Comment by maras93** [comment]
I think now it's ok :)

**Pull request merged**